server/events/apply_command_runner.go

```diff
@@ -2,14 +2,49 @@ package events
 
 import (
 	"github.com/runatlantis/atlantis/server/events/db"
+	"github.com/runatlantis/atlantis/server/events/locking"
 	"github.com/runatlantis/atlantis/server/events/models"
 	"github.com/runatlantis/atlantis/server/events/vcs"
 )
 
+//go:generate pegomock generate -m --use-experimental-model-gen --package mocks -o mocks/mock_apply_command_locker.go ApplyCommandLocker
+
+type ApplyCommandLocker interface {
+	IsDisabled(ctx *CommandContext) bool
+}
+
+func NewApplyCommandLocker(
+	applyLockChecker locking.ApplyLockChecker,
+	disableApply bool,
+) *DefaultApplyCommandLocker {
+	return &DefaultApplyCommandLocker{
+		ApplyLockChecker: applyLockChecker,
+		DisableApply:     disableApply,
+	}
+}
+
+type DefaultApplyCommandLocker struct {
+	ApplyLockChecker locking.ApplyLockChecker
+	DisableApply     bool
+}
+
+// IsDisabled returns true if there is a global apply command lock or
+// DisableApply flag is set to true
+func (a *DefaultApplyCommandLocker) IsDisabled(ctx *CommandContext) bool {
+	lock, err := a.ApplyLockChecker.CheckApplyLock()
```

Comment:
Looks like command context is barely used here. The logger embedded in there is useful if we want to debug per-pull issues. However, the global apply lock fetching error doesn't seem necessary to log at that level, you can just keep that logger embedded in the struct.


If you do the above, you can just use a single interface delegation pattern for:
1) fetching the global lock
2) comparing the result of that to the global server config flag.